src/dynacli/dynacli.py

```diff
@@ -135,6 +136,15 @@ def _is_public(name: str) -> bool:
     return not name.startswith("_")
 
 
+def _is_origin_from_search_path(name: str, search_path: List[str]) -> bool:
+    paths = [path_ + name.replace(".", "/") for path_ in search_path]
```

Comment:
Could we prepare it once for all in __init__. Performing the same operation over and over seems to be wasteful

Reply:
The `name` here is the actual imported `module`'s `__module__` and the imported package's `__package__`, so we do not have that information beforehand. We grab the module from the `_current_package.__dict__` and it differs at each feature/package.

Reply:
This solution is too wasteful. You need to take ```__file__ of the module and check if it starts with one of predefined search_path * root package combination prepared in advance.

Reply:
No real file io

Reply:
With imported functions I got:

```
AttributeError: 'function' object has no attribute '__file__'
```
Will push the workaround.

Reply:
This should be: `module.__code__.co_filename` -> for function object.